### PR TITLE
Code Signing on Windows, Try 2

### PR DESCRIPTION
**Pull request by mook-as**
This is a redo of #365, now that we have access to an actual code signing certificate and have a better idea of what we're doing.  Also, from the previous PR I've figured out that we can just manually sign the executables ahead of time to work around the electron-builder issue where it doesn't sign them itself.

Hopefully the [documentation](https://github.com/mook-as/rd/blob/code-signing-v2/docs/development/signing.md) provides enough information to test code signing.  I've rolled the previous suggestion in where they still make sense.

There should still be a follow-up for macOS notarization.

### CODE REVIEW

File: scripts/lib/sign-win32.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    throw new Error(`CSC_FINGERPRINT environment variable not set; required to pick signing certificate.`);
+  }
+
+  const configText = await fs.promises.readFile('electron-builder.yml', 'utf-8');
+  const config = yaml.safeLoad(configText) as ElectronBuilderConfiguration;
+
+  config.win ??= {};
+  defaults(config.win, DEFAULT_WINDOWS_CONFIG);
+
+  // Sign indiviaul files.  See https://github.com/electron-userland/electron-builder/issues/5968
```
Comment (ericpromislow) [inline scripts/lib/sign-win32.ts:53]: tyop: "individual"

**Review by ericpromislow — CHANGES_REQUESTED** [review]
```
npm run sign -- '.\dist\Rancher Desktop-0.3.0-80-g92d000c-win.zip'
```
ran to completion with the FINGERPRINT setting from above. Apart from the typo in the comment the code looks good.

Rather than test this will this be used by the CI, so I can verify that I no longer get the untrusted-software prompt on a release build?

**Comment by mook-as** [comment]
(I will get to the fixes in a bit.)

No, the code signing _will not_ be used by CI; CI doesn't have access to our certificates:
- CI isn't under our control (they're GitHub-managed, Azure-hosted VMs)
- We don't really want the certificates to be always online; instead, we keep it airgapped.  Ideally, we'd have a dedicated machine (that is always offline) for signing.

**Comment by ericpromislow** [comment]
Doh, wasn't thinking and was hoping that the CI would do the grunt work. But obviously not via github actions. We do have a CI for signing, it's just not automated (or very continuous, for that matter).

**Review by ericpromislow — APPROVED** [review]
Approving it again....

**Pull request merged**